.github/workflows/argilla.docs.yml

```diff
@@ -64,3 +65,11 @@ jobs:
 
       - run: pdm run mike deploy ${{ github.ref_name }}
         if: startsWith(github.ref, 'refs/tags/')
+
+      - name: Extract branch name
+        id: extract_branch_name
+        shell: bash
+        run: echo "branch_name=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" | sed 's/\//_/g'  >> $GITHUB_OUTPUT
+
+      - run: pdm run mike deploy ${{ steps.extract_branch_name.outputs.branch_name }} --prop-set hidden=true --push
```

Comment:
We can start with this, but I think we need one more thing for a useful MVP for us and non-technical people:
- more active way to publish the public URL / even if it is hidden from versioning, otherwise people still need to look into copy-pasting the branch name, replacing slashes with underscores and then pasting it after the source URL. This can be done with something, like this: https://github.com/marketplace/actions/create-or-update-comment or https://github.com/marketplace/actions/add-pr-comment

And one for a follow-up:
- cleanup workflow when releasing a new tag, otherwise we overload the [gh-page branch](https://github.com/argilla-io/argilla/tree/gh-pages) with different builds for each version of `docs/ `

Reply:
@frascuchon This was handled in my more recent commits, feel free to take a look.